### ISSUE TITLE
[EXP] Evaluate Deep & Cross Network (DCN) Performance

### ISSUE THREAD

**Issue by twndus** [twndus/YelpRecommendation#18]
## Background
- Evaluating DCN model's performance on Yelp dataset using W&B sweep function

## To do
- [x] make sweep config (HP search by bayesian)
   - common configs: learning rate, batch_size, optimizer, weight_decay
   - dcn-specific configs: embed_size, cross_order, hidden_dims
- [x] run sweep agent
- [ ] log best HP and best performance (best metric: MAP@10)
- [ ] ablation study
   - cross-only, cross_order
   - deep-only, hidden_dims

※ epoch: 100, early stopping: 5 (MAP@10) fixed

**Comment by twndus** [comment]
I've adjusted the embedding sizes in our implementation, moving from a hyper-parameter to constants based on each data category's cardinality, as described in section 4.2 of the DCN paper.

[DCN](https://arxiv.org/pdf/1708.05123) 4.2 Implementation Details

> Data processing and embedding. Real-valued features are normalized by applying a log transform. For categorical features, we embed the features in dense vectors of dimension 6×(category cardinality)1/4 . Concatenating all embeddings results in a vector of dimension 1026.

e.g) total number of users: 19855, embed dim:  71 (6 * (19855 ** 0.25)), total input: 197 (including 3 other features)

❗ initial losses were reduced after applying the above formula.
<img width="416" alt="image" src="https://github.com/twndus/YelpRecommendation/assets/95452963/d9fd1de5-f5bc-4a19-b708-38e1e51d9a9b">

**Comment by GangBean** [comment]
- Based on the experiments conducted so far, the performance does not match that of CDAE or MF.
- It does not seem to be an implementation issue.
- Plan to implement other models that use features for comparison and re-run experiments.
